service/storage_service.cc

```diff
@@ -3266,6 +3268,16 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
     } ();
 
     co_await _gossiper.wait_for_gossip_to_settle();
+
+    // This is the moment when the locator::topology has gathered information about other nodes
+    // in the cluster -- either through gossiper, or by loading it from disk -- so it's safe
+    // to start the hint managers.
```

Comment:
I think this logic is valid in gossip topology mode, but not in raft topology mode.
@piodul please review the change

Reply:
At this point, the group0 server was started, and we apply all of the committed entries when a raft server is started. A decision whether to store a hint or not is done locally, so I think this is sufficient and a read barrier is not necessary (if that was what you meant by this logic being invalid).

Reply:
Right, thanks.

Reply:
@dawmd - please resolve comment threads (such as this one?)